src/error.rs

```diff
@@ -0,0 +1,38 @@
+use derive_more::Display;
+use std::{error::Error, num::NonZeroI32};
+
+/// Error types emitted by `pn` itself.
+#[derive(Debug, Display)]
+pub enum PnError {
+    /// Script not found when running `pn run`.
+    #[display(fmt = "Missing script: {name}")]
+    MissingScript { name: String },
+    /// Script ran by `pn run` exits with non-zero status code.
+    #[display(fmt = "Command failed with exit code {status}")]
+    ScriptError { name: String, status: NonZeroI32 },
+    /// Subprocess finishes but without a status code.
+    #[display(fmt = "Command {command:?} has ended unexpectedly")]
+    UnexpectedTermination { command: String },
+    #[display(fmt = "--workspace-root may only be used in a workspace")]
+    NotInWorkspace,
+    /// Other errors.
+    #[display(fmt = "{error}")]
+    Other { error: Box<dyn Error> },
```

Comment:
`Other` was used because this code was a prototype. But if you want to seriously develop this, I think one day we should divide this error variant into multiple specific error variants, such as `ManifestReadingError(io::Error)`, `ManifestParsingError(serde::Error)`, and `FindUpError(io::Error)` for example.

If you don't want to deal with this right now, you can move this little suggestion of mine into a separate issue.

Reply:
Issue created https://github.com/pnpm/pn/issues/16